### PR TITLE
Ensure deletion of snapshot CR when it is gone from engine

### PR DESCRIPTION
**Pull request by ejweber**
longhorn/longhorn#6298

### CODE REVIEW

File: controller/snapshot_controller.go [hunk 1]
```diff
@@ -335,6 +336,14 @@ func (sc *SnapshotController) reconcile(snapshotName string) (err error) {
...
 		}()
 
+		// If the snapshot no longer exists in the engine, allow its CR to be deleted.
+		if _, ok := engine.Status.Snapshots[snapshot.Name]; !ok {
+			if err = sc.handleAttachmentTicketDeletion(snapshot); err != nil {
+				return err
+			}
+			return sc.ds.RemoveFinalizerForSnapshot(snapshot)
+		}
+
```
Comment (PhanLe1010) [inline controller/snapshot_controller.go:346]: This is a bit dangerous. Sometimes the `engine.Status.Snapshot` info might be clear (live migration, live upgrade) but the snapshot still exists on disk. We should not check with the engine process before removing the finalizer

Comment (ejweber) [reply controller/snapshot_controller.go:346]: I think I agree in principle, but I'm not sure I see where we were previously doing so. Do you mean on 357 when we verify the engine is running?

Comment (PhanLe1010) [reply controller/snapshot_controller.go:346]: Previously, we make sure engine is running, issue the snapshotdelete request to engine, then check the engine.status.snapshot

This make sure that at least we issue the snapshot delete command to engine process successfully 

But this PR might not attempt that snapshotdelete request

Comment (ejweber) [reply controller/snapshot_controller.go:346]: We discussed this offline and I am now in agreement. Closing this PR.

**Pull request closed without merging**